src/segwit.rs

```diff
@@ -100,6 +102,19 @@ pub fn encode(
     Ok(buf)
 }
 
+/// Decodes a bech32 encoded string.
+///
+/// Reads a string from reader and then decodes it (see [`crate::decode`]).
+#[cfg(feature = "std")]
+#[inline]
+pub fn decode_from_reader<R: std::io::Read>(
+    r: &mut R,
+) -> Result<(Hrp, Fe32, Vec<u8>), DecodeFromReaderError> {
+    let mut s = String::new();
+    r.read_to_string(&mut s)?;
+    Ok(decode(&s)?)
```

Comment:
In 68c4dfd2666ff390cc10d14adc7d5302667e3f20:

I'll go ahead and ACK this since it's not API-visible, but the string allocation (and UTF-8 checks and everything that comes with it) are all unnecessary here. You should be able to just use the `bytes()` iterator on the `Read`, adapt it wiht `char::from` to get characters, then use the iterator API.

Reply:
I like it! I'll have go.

Reply:
I had a bit of a play with this, unless I'm missing something:
- The iterator API (you mean `iter.rs`, right?) is for encoding not decoding
- The API in `decode.rs` could be changed to enable decoding from a reader but its non-trivial because all the decoding types (eg, `UncheckedHrpstring`) have lifetimes that expect a string somewhere to reference.

We could read into a buffer because a bech32 string is max 90 chars, then parse it out however we like.

Or am I totally missing something?

Reply:
No, you may be correct. Several iterations ago I had an idea for decoding through the iterator API and I guess that got dropped.

I think it's fine to just drop this function then. I don't have a clear usecase for it and it doesn't even work with nostd.

Reply:
Removed, both in `lib.rs` and `segwit.rs`